hardhat.config.js/tasks.js

```diff
@@ -43,3 +43,11 @@ task(TASK_COMPILE)
 
     await runSuper();
   });
+
+task('coverage').setAction(async function (args, hre, runSuper) {
```

Comment:
i saw the build failing so it looks like it's working .


My only concern here:

Why is this part of the coverage stage and not another stage?

Compilation fails normally in checkout and install right?

Reply:
You can try changing the contract with something you know that is a warning (size of the contract) and try running compile. Now it will be an error and it will fail.

for the coverage it needs to have warning instead of the error because of all the mocks and tests. It doesn’t even run the tests if I leave them as error, so only for coverage we will use warnings.

Reply:
ok understood so basically you are preventing the breakage of this stage so it only fails at compile